opentasks-provider/src/androidTest/java/org/dmfs/provider/tasks/TaskProviderTest.java

```diff
@@ -230,12 +369,51 @@ public void testInsertWithStartAndDuration()
                         new EqArg(Instances.INSTANCE_START, start.getTimestamp()),
                         new EqArg(Instances.INSTANCE_DUE, start.addDuration(duration).getTimestamp()),
                         new EqArg(Instances.INSTANCE_DURATION, durationMillis),
+                        new EqArg(Instances.INSTANCE_START_SORTING, start.shiftTimeZone(TimeZone.getDefault()).getInstance()),
+                        new EqArg(Instances.INSTANCE_DUE_SORTING, start.addDuration(duration).shiftTimeZone(TimeZone.getDefault()).getInstance()),
+                        new EqArg(Instances.INSTANCE_ORIGINAL_TIME, start.getTimestamp()),
                         new EqArg(Tasks.TZ, "UTC")
                 ))
         ));
     }
 
 
+    /**
+     * Create task with start and duration, check datetime values including generated due.
+     */
+    @Test
+    public void testInsertWithStartAndDurationChangeTimeZone()
+    {
+        RowSnapshot<TaskLists> taskList = new VirtualRowSnapshot<>(new LocalTaskListsTable(mAuthority));
+        RowSnapshot<Tasks> task = new VirtualRowSnapshot<>(new TaskListScoped(taskList, new TasksTable(mAuthority)));
+
+        DateTime start = DateTime.now();
+        Duration duration = Duration.parse("PT1H");
+        long durationMillis = duration.toMillis();
+        DateTime startNew = start.shiftTimeZone(TimeZone.getTimeZone("America/New_York"));
+
+        assertThat(new Seq<>(
+                new Put<>(taskList, new EmptyRowData<TaskLists>()),
+                new Put<>(task, new TimeData(start, duration)),
+                // update the task with a the same start in a different time zone
+                new Put<>(task, new TimeData(startNew, duration))
+
+        ), resultsIn(mClient,
+                new Assert<>(task, new TimeData(startNew, duration)),
+                // note that, apart from the time zone, all values stay the same
```

Comment:
Why is it that the values apart from timezone don't have to be updated in the Instance table?

Reply:
because the instances table contains only timestamps (which are always in UTC, so they don't change when you shift the event to a different time zone keeping the absolute time constant) and sorting values which only depend on the local time zone of the device and won't change neither as along as you keep the absolute time constant.

Reply:
Also note that the time zone is not even stored in the instances table and there is no need to test it separately because it's already covered by the previous assertion. So if you just shift a task to a different time zone keeping the absolute time constant, the instances table won't change at all.

Reply:
Thanks, I understand it now from your other comment, too.